github/data_source_github_collaborators.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceGithubCollaborators() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGithubCollaboratorsRead,
+
+		Schema: map[string]*schema.Schema{
+			"owner": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"repository": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"affiliation": {
+				Type: schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					"all",
+					"direct",
+					"outside",
+				}, false),
+				Optional: true,
+				Default:  "all",
+			},
+			"collaborator": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"login": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"html_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"followers_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"following_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"gists_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"starred_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"subscriptions_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"organizations_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"repos_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"events_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"received_events_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"site_admin": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"permission": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceGithubCollaboratorsRead(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*Organization).client
+	ctx := context.Background()
+
+	owner := d.Get("owner").(string)
+	repo := d.Get("repository").(string)
+	affiliation := d.Get("affiliation").(string)
+
+	options := &github.ListCollaboratorsOptions{
+		Affiliation: affiliation,
+		ListOptions: github.ListOptions{
+			PerPage: maxPerPage,
+		},
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", owner, repo, affiliation))
+	d.Set("owner", owner)
+	d.Set("repository", repo)
+	d.Set("affiliation", affiliation)
+
+	totalCollaborators := make([]interface{}, 0)
+	for {
+		collaborators, resp, err := client.Repositories.ListCollaborators(ctx,
```

Comment:
Maybe merge these two lines?
```suggestion
		collaborators, resp, err := client.Repositories.ListCollaborators(ctx, owner, repo, options)
```